src/main/java/com/aws/greengrass/clientdevices/auth/api/UseCases.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.api;
+
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.dependency.Context;
+import com.aws.greengrass.util.CrashableFunction;
+
+
+public class UseCases {
+    private final Context context;
+    private static UseCases instance;
+
+    // Delegates to CrashableFunction but provides a domain rich alias
+    public interface UseCase<R, D, E extends Exception> extends CrashableFunction<D, R, E> {}
+
+    private UseCases(Context context) {
+        this.context = context;
+    }
+
+    public static void init(Topics topics) {
```

Comment:
Add the call to `init` in the service entry?

Reply:
Yeah, I intended to do that on the next PR but I can do it in this one